tests/platform_tests/test_link_down.py

```diff
@@ -0,0 +1,210 @@
+    hostname = duthost.hostname
+    dut_ip = duthost.mgmt_ip
+    logger.info('waiting for ssh to drop on {}'.format(hostname))
+    res = localhost.wait_for(host=dut_ip,
+                             port=SONIC_SSH_PORT,
+                             state='absent',
+                             search_regex=SONIC_SSH_REGEX,
+                             delay=delay,
+                             timeout=timeout,
+                             module_ignore_errors=True)
+
+    if res.is_failed or ('msg' in res and 'Timeout' in res['msg']):
+        if reboot_res.ready():
+            logger.error('reboot result: {} on {}'.format(reboot_res.get(), hostname))
+        raise Exception('DUT {} did not shutdown'.format(hostname))
+
+    if not wait_for_ssh:
+        return
+
+def test_link_down_on_sup_reboot(duthosts, localhost, enum_supervisor_dut_hostname, 
+                                conn_graph_facts, duts_running_config_facts, 
+                                fanouthosts, tbinfo, xcvr_skip_list):
+    if len(duthosts.nodes) == 1:
+        pytest.skip("Skip single-host dut for this test")
+        
+    duthost = duthosts[enum_supervisor_dut_hostname]
+
+    if 't2' not in tbinfo['topo']['name']:
+        pytest.skip("Skip for non-t2 supervisor card")
+   
+    # Before test, check all interfaces and services are up on all linecards
+    check_interfaces_and_services_all_LCs(duthosts, conn_graph_facts, xcvr_skip_list)
+    
+    duts_and_ports = multi_duts_and_ports(duthosts)
+    fanouts_and_ports = fanout_hosts_and_ports(fanouthosts, duts_and_ports)
+    
+    # Reboot RP should reboot both RP&LC, should detect all links on all linecards go down
+    logger.info("Rebooting RP {} and checking all linecards' interfaces".format(duthost.hostname))
+
+    hostname = duthost.hostname
+    pool = ThreadPool()
+    
+    def execute_reboot_command():
+        return duthost.command("reboot")
+
+    dut_datetime = duthost.get_now_time()
+    reboot_res = pool.apply_async(execute_reboot_command)
+    wait_for_shutdown(duthost, localhost, 0, MAX_TIME_TO_REBOOT, reboot_res, True)
+
+    # RP doesn't have any interfaces, check all LCs' interfaces
+    links_down_on_all_LC(duthosts, localhost, fanouts_and_ports)
+
+    time.sleep(MAX_TIME_TO_REBOOT)
+
+    logger.info('reboot finished on {}, reboot started on {}'.format(hostname, dut_datetime))
+    dut_uptime = duthost.get_up_time()
+    logger.info('DUT {} up since {}'.format(hostname, dut_uptime))
+    assert float(dut_uptime.strftime("%s")) > float(dut_datetime.strftime("%s")), "Device {} did not reboot".format(hostname)
+
+
+def test_link_down_on_host_reboot(duthosts, localhost, enum_rand_one_per_hwsku_frontend_hostname, 
+                                 duts_running_config_facts, conn_graph_facts, 
+                                 fanouthosts, xcvr_skip_list, tbinfo):
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
```

Comment:
can use the fixture enum_frontend_dut_hostname instead of enum_rand_one_per_hwsku_frontend_hostname to test all the linecard skus present on the chassis